artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -647,66 +649,121 @@ public QueueBinding updateQueue(QueueConfiguration queueConfiguration) throws Ex
                }
             }
 
-            //atomic update
-            if (queueConfiguration.getMaxConsumers() != null && queue.getMaxConsumers() != queueConfiguration.getMaxConsumers().intValue()) {
+            // atomic update, reset to defaults if value == null
+            // maxConsumers
+            final int newMaxConsumers = queueConfiguration.getMaxConsumers() == null
+                    ? ActiveMQDefaultConfiguration.getDefaultMaxQueueConsumers()
+                    : queueConfiguration.getMaxConsumers();
+            if (queue.getMaxConsumers() != newMaxConsumers) {
                changed = true;
-               queue.setMaxConsumer(queueConfiguration.getMaxConsumers());
+               queue.setMaxConsumer(newMaxConsumers);
             }
-            if (queueConfiguration.getRoutingType() != null && queue.getRoutingType() != queueConfiguration.getRoutingType()) {
+            // routingType
+            final RoutingType newRoutingType = queueConfiguration.getRoutingType() == null
```

Comment:
So to get defaults, should be using the address settings.

Theres actually a common method that does this thats used else where, as its quite common

     QueueConfigurationUtils.applyDynamicQueueDefaults(queueConfiguration, addressSettingsRepository.getMatch(queueConfiguration.getAddress().toString()));

This then will correctly set defaults on everything in the queueConfiguration

Reply:
e.g. you could just do this once before the section of atomic updates.